File: AULA_08/JAVA/Raiz.java
/* Raiz.java
 * Encontrar a raiz n do número x
 * utilizando o método de Newton
 *
 * Entrada: y,n (double) 
 * Saída: raiz
 *
 * Autor: Fabrício Olivetti de França
 * Disciplina Processamento da Informação
 * Universidade Federal do ABC
 */


import java.util.Scanner;

class Raiz{

	// avalia a função f(x) = x^n - y
	public static double f( double x, double y, double n ){
		return Math.pow(x,n)-y;
	}

	// avalia f'(x) = n*x^(n-1)
	public static double df( double x, double n ){
		return Math.pow(x,n-1)*n;
	}

	public static void main(String[] args) {
		Scanner leitor = new Scanner(System.in);
		double y, n;
		y = leitor.nextDouble();
		n = leitor.nextDouble();

		double x = 1, xAnt = 2;

		while( Math.abs(x-xAnt) > 0.01 ){ // até que altere apenas a segunda casa decimal
			xAnt = x;
			x -= f(x,y,n)/df(x,n);  // método de Newton
		}
		System.out.println("A raíz " + n + " de " + y + " = " + x);

	}
}

File: AULA_08/Python/Raiz.py
# -*- coding: cp1252 -*-
'''
Raiz.py
 * Encontrar a raiz n do número x
 * utilizando o método de Newton
 *
 * Entrada: y,n (double) 
 * Saída: raiz
 *
 * Autor: Fabrício Olivetti de França
 * Disciplina Processamento da Informação
 * Universidade Federal do ABC
'''

def f( x,y,n ):
    return x**n - y

def df( x,n ):
    return n*(x**(n-1))

if __name__ == "__main__":

    y = float(raw_input("y = "))
    n = float(raw_input("n = "))
    x = 1.0
    xAnt = 2.0
    while abs(x-xAnt) > 0.01:
        xAnt = x
        x -= f(x,y,n)/df(x,n)
    print "Raíz",n,"de",y,"=",x
        
